### DISCUSSION TITLE
How to display a shape more than once with different templates?

### DISCUSSION THREAD

**Discussion by DrewBrasher** [OrchardCMS/OrchardCore#10175] [Q&A]
I have a content type called **BoardPacket** that has a **ListPart**.  I would like to display just the outline of the items in the ListPart first and then display the full content of the items.  I created a liquid template for the **BoardPacket** called `Content-BoardPacket.liquid` and two razor templates for the ListPart called `BoardPacket-ListPart-Full.cshtml` and `BoardPacket-ListPart-Agenda.cshtml`.  I have tried this in the `Content-BoardPacket.liquid` template :
```
      {% shape_add_alternates Model.Content.ListPart "BoardPacket__ListPart__Agenda" %}
      {{ Model.Content.ListPart | shape_render }}
      
      {% shape_clear_alternates Model.Content.ListPart %}
      {% shape_type Model.Content.ListPart "BoardPacket__ListPart__Full" %}
      {{ Model.Content.ListPart | shape_render }}
```
from https://github.com/OrchardCMS/OrchardCore/discussions/10165#discussioncomment-1240388
but it renders the Agenda template twice instead of rendering both templates.

**Comment by dodyg** [comment]
@DrewBrasher did you ever figure this out?

**Reply by DrewBrasher** [reply]
Not really, I ended up just making one ListPart razor template and then looping through the items in the list part twice to accomplish what I wanted. Probably not the best solution but it worked.

**Reply by dodyg** [reply]
Yeah Shape and the OC rendering mechanism can be a bit of a black art to me. It took 20 posts for me to figure out how to render Bag properly https://github.com/OrchardCMS/OrchardCore/discussions/11278
